sound/soc/sof/topology.c

```diff
@@ -1704,15 +1735,15 @@ static int sof_widget_load_pga(struct snd_soc_component *scomp, int index,
 
 	ret = sof_parse_tokens(scomp, volume, volume_tokens,
 			       ARRAY_SIZE(volume_tokens), private->array,
-			       le32_to_cpu(private->size));
+			       le32_to_cpu(private->size), &read_bytes);
 	if (ret != 0) {
 		dev_err(scomp->dev, "error: parse volume tokens failed %d\n",
 			private->size);
 		goto err;
 	}
 	ret = sof_parse_tokens(scomp, &volume->config, comp_tokens,
 			       ARRAY_SIZE(comp_tokens), private->array,
-			       le32_to_cpu(private->size));
+			       le32_to_cpu(private->size), &read_bytes);
```

Comment:
@juimonen whats the purpose of read_bytes here? How is it used?

Reply:
so the read_bytes will give you the number of how many bytes from the beginning of the data area the last array was found and parsed. So you can parse next array from that point and not start from beginning again and again... You can see the example in the demux parsing in this pr.